aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2545,14 +2592,20 @@ Tensor mexp_impl(
   bool compute_highest_degree_approx = false
 ) {
   auto res = at::empty_like(a);
+  res.index_put_({at::indexing::Slice(), 0, 0}, std::numeric_limits<double>::quiet_NaN());
```

Comment:
This doesn't make sense.

Reply:
NaN check is mainly for preventing undefined behaviors while the inputs have NaN values, however, it can affect the performance for small batches.
In previous versions, the inputs' `nan` value will be gone after we perform `mexp`, which causes the NN to output the undefined "good" result rather than the obviously wrong result that includes NaN, or raise errors.

This line is for keeping the NaN thing.

Reply:
let's roll back all the nan-related fixes and keep this PR just about performance.
Then, you can open a second PR with a fix and a regression test fixing any NaN issue.

Reply:
Ok, it should be.